pkg/cmd/opts/kustomize.go

```diff
@@ -0,0 +1,54 @@
+package opts
+
+import (
+	"github.com/blang/semver"
+	"github.com/jenkins-x/jx/pkg/log"
+	"github.com/jenkins-x/jx/pkg/versionstream"
+	"github.com/pkg/errors"
+)
+
+// EnsureKustomize ensures kustomize is installed
+func (o *CommonOptions) EnsureKustomize() error {
+	version, err := o.Kustomize().Version()
+
+	if err == nil && version != "" {
+		// get the stable jx supported version of kustomize to be install
+		versionResolver, err := o.GetVersionResolver()
+		if err != nil {
+			return errors.Wrapf(err, "Unable to get version resolver for jenkins-x-versions")
+		}
+
+		stableVersion, err := versionResolver.StableVersion(versionstream.KindPackage, "kustomize")
+		if err != nil {
+			return errors.Wrapf(err, "Unable to get stable version from the jenkins-x-versions for github.com/%s/%s %v ", "kubernetes-sigs", "kustomize", err)
+		}
+
+		currVersion, err := semver.Make(version[12:17])
```

Comment:
That seems wrong and fragile. where is '12:17' coming from. When I run `kustomize version --short`, I'll get `{3.5.4  2020-01-17T14:28:58+00:00  }`, so if anything it should be something like '1:6', right?

Right now I get test failures due to that.

I think working with `kustomize version` might be more understandable as well. It returns: 
```
{Version:3.5.4 GitCommit:3af514fa9f85430f0c1557c4a0291e62112ab026 BuildDate:2020-01-17T14:28:58+00:00 GoOs:darwin GoArch:amd64}
```
I'd probably use a regexp like `^*Version:(.*) .*` to capture the version and take it from there. It makes it more explicit what you are expecting at this point. If you just take a slice out of string you depend much more on the exact string representation. Also if doing though, I'd add a comment about what you expect the _version_ to be in order for you to use the slice '12:17'.

Reply:
And should we really just continue if there is an error?

Reply:
[0-9]+\.[0-9]+\.[0-9]

Reply:
[\d]+.[\d]+.[\d]